pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -765,6 +766,24 @@ private void writeMetadata()
               "Invalid segment start/end time: %s (in millis: %s/%s) for time column: %s, must be between: %s",
               timeInterval, timeInterval.getStartMillis(), timeInterval.getEndMillis(), timeColumnName,
               TimeUtils.VALID_TIME_INTERVAL);
+        } else {
+          long now = System.currentTimeMillis();
```

Comment:
We don't want to change the start/end time within the segment metadata, instead we want to enhance the time management code to gracefully handle the segments with invalid time values.
Suggest using a separate PR for the time management change, and only keep the sample value check in this PR.